extmod/modlwip.c

```diff
@@ -1033,7 +1033,7 @@ STATIC mp_obj_t lwip_socket_sendall(mp_obj_t self_in, mp_obj_t buf_in) {
             break;
         }
         case MOD_NETWORK_SOCK_DGRAM:
-            mp_not_implemented("");
+            mp_raise_NotImplementedError("");
```

Comment:
Your commit message doesn't mention that you rename mp_not_implemented() to mp_raise_NotImplementedError(). And I'd say that's too much for one patch. This renaming would rather go to a separate RFC pull request.